### PR TITLE
Added pass for xdebug requests

### PR DESCRIPTION
**Pull request by aepod**
Dimitar Ivanov provided the code for this in the community engineering slack.

### Description
This fixes the varnish+xdebug issue.

### Fixed Issues (if relevant)
1. https://github.com/magento/magento-cloud-docker/issues/110

### Manual testing scenarios
1. Enable varnish and xdebug, and do a step debug.

### Contribution checklist
 - [X] Pull request has a meaningful description of its purpose
 - [X] All commits are accompanied by meaningful commit messages

### CODE REVIEW

**Comment by shiftedreality** [comment]
Thank you for your contribution!

Changing target to 1.0 to include into the upcoming release.

**Comment by shiftedreality** [comment]
Created task https://magento2.atlassian.net/browse/MAGECLOUD-4873

**Comment by joeshelton-wagento** [comment]
Possible to only check for the existence of the cookie, as opposed to the value "PHPSTORM"? This would make it more versatile and agnostic.

Also, some people use a GET or POST parameter "XDEBUG_SESSION_START". https://xdebug.org/docs/remote

**Comment by aepod** [comment]
https://devdocs.magento.com/guides/v2.3/cloud/docker/docker-development-debug.html
This only provides instructions for PHPStorm, I think for now this PR is fine, it matches with given instructions. We can improve on this later, and add support for other IDE, but as far as I can see this matches with documentation and should be good to go for now.

Please let me know if it needs to be adjusted.

**Review by shiftedreality — APPROVED** [review]

**Review by oshmyheliuk — DISMISSED** [review]

**Comment by shiftedreality** [comment]
@aepod could you please add the same change to Varnish 4.0 image?

File: images/varnish/6.2/etc/default.vcl [hunk 1]
```diff
@@ -28,6 +28,11 @@ sub vcl_recv {
         return (synth(200, "Purged"));
     }
 
+    # Bypass debug session
+    if (req.http.Cookie ~ "(^|;\s*)(XDEBUG_SESSION=PHPSTORM)(;|$)") {
```
Comment (shiftedreality) [inline images/varnish/6.2/etc/default.vcl]: Please add Varnish 4.0 support here

**Review by shiftedreality — CHANGES_REQUESTED** [review]

**Comment by shiftedreality** [comment]
Grooming comment: Make session variable broader to include more variety of IDEs

**Comment by aepod** [comment]
PR has been updated, it now just checks for the cookie (not contents) also it is added to both v4 and v6 varnish config.

**Review by shiftedreality — APPROVED** [review]

**Review by oshmyheliuk — APPROVED** [review]

**Comment by mveeramneni** [comment]
QA Approved

**Comment by shiftedreality** [comment]
@aepod will you add doc update mentioning this change or should I create a task internally?

**Pull request merged**